vllm/core/scheduler.py

```diff
@@ -92,7 +92,10 @@ def abort_seq_group(self, request_id: Union[str, Iterable[str]]) -> None:
             request_id = (request_id, )
         request_ids = set(request_id)
         for state_queue in [self.waiting, self.running, self.swapped]:
-            for seq_group in state_queue:
+            # We need to reverse the list as we are removing elements
+            # from it as we iterate over it. If we don't do it,
+            # indices will get messed up and we will skip over elements.
+            for seq_group in reversed(state_queue):
```

Comment:
I think the way forward here would be to refactor it completely so that we don't need to iterate over the lists in the first place. Perhaps we can use ordered dicts?